infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -313,4 +373,43 @@ private CompletableFuture<LogReplicationEntryMsg> timeEntrySend(Timer.Sample sam
                     return future;
                 }).orElse(entryFuture);
     }
+
+
+    /**
+     * Utility class to provide a configurable linearly increasing sleep mechanism.
+     *
+     */
+    public static class LinearBackoffRetry {
+        private final long initialSleepTime;
+        private final long maxSleepTime;
+        private final long sleepTimeIncrement;
+
+        @Getter
+        private long currentSleepTime;
+
+        public LinearBackoffRetry(long maxSleepTime, long initialSleepTime, long sleepTimeIncrement) {
+            this.maxSleepTime = maxSleepTime;
+            this.initialSleepTime = initialSleepTime;
+            this.sleepTimeIncrement = sleepTimeIncrement;
+            this.currentSleepTime = initialSleepTime;
+        }
+
+        public void backoff() {
+            try {
+                log.trace("ACKs timing out on sink, resending messages in {} milliseconds...", currentSleepTime);
+                TimeUnit.MILLISECONDS.sleep(currentSleepTime);
+            } catch (Exception e) {
+                Thread.currentThread().interrupt();
```

Comment:
why do we interrupt the current thread here?

Reply:
Had this here from when I had the wait set to some order of minuets, so the interrupted status is preserved and the next call to sleep exits without waiting until the interrupt flag is reset which would come when `processAcks()` hits a InterruptedException on the next blocking call and clears it implicitly. Can remove this for now since the logic has changed since and reinterrupting the thread doesn't appear to have any benefits